src/selectedtests/task_mappings/mappings.py

```diff
@@ -0,0 +1,367 @@
+    return []
+
+
+def _get_associated_module(version: Version, module_name: str) -> ManifestModule:
+    """
+    Get the associated module for the given version.
+
+    :param version: The version to get the module from.
+    :param module_name: The name of the module to get.
+    :return: The module that was asked for. Can return None if that module wasn't found.
+    """
+    modules = version.get_manifest().modules
+    return modules.get(module_name)
+
+
+def _get_diff(repo: Repo, cur_revision: str, prev_revision: str) -> DiffIndex:
+    """
+    Create a diff between two revisions.
+
+    :param repo: The repo that contains the two given revisions.
+    :param cur_revision: The child revision.
+    :param prev_revision: The parent revision.
+    :return: The diff between the two given revisions.
+    """
+    cur_commit = repo.commit(cur_revision)
+    parent = repo.commit(prev_revision)
+
+    return cur_commit.diff(parent)
+
+
+def _map_tasks_to_files(changed_files: List[str], flipped_tasks: Dict, task_mappings: Dict):
+    """
+    Map the flipped tasks to the changed files found in this version. Mapping will be done in \
+    place in the dictionary given in the task_mappings parameter.
+
+    :param changed_files: List of the files that changed in this version.
+    :param flipped_tasks: Dictionary that contains the build variants as keys and the list of tasks
+     that changed in that variant as the value.
+    :param task_mappings: Where the mappings will be stored. New mappings will be added to this
+     dictionary in place.
+    """
+    for file_name in changed_files:
+        task_mappings_for_file = task_mappings.setdefault(
+            file_name, {TASK_BUILDS_KEY: {}, SEEN_COUNT_KEY: 0}
+        )
+        task_mappings_for_file[SEEN_COUNT_KEY] = task_mappings_for_file[SEEN_COUNT_KEY] + 1
+        build_mappings = task_mappings_for_file[TASK_BUILDS_KEY]
+        for build_name in flipped_tasks:
+            builds_to_task_mappings: Dict[str:Dict] = build_mappings.setdefault(build_name, {})
+            for cur_task in flipped_tasks.get(build_name):
+                cur_flips_for_task = builds_to_task_mappings.setdefault(cur_task, 0)
+                builds_to_task_mappings[cur_task] = cur_flips_for_task + 1
+
+
+def _init_repo(temp_dir, org_name: str, repo_name: str, branch: str) -> Repo:
+    """
+    Create the given repo in the given directory and checkout the given branch.
+
+    :param temp_dir: The place where to clone the repo to or check for it already existing.
+    :param org_name: The org name in github that owns the repo.
+    :param repo_name: The name of the repo to clone.
+    :param branch: The branch to checkout in the repo.
+    :return: An Repo instance that further git operations can be done on.
+    """
+    repo_path = os.path.join(temp_dir, repo_name)
+    if os.path.exists(repo_path):
+        repo = Repo(repo_path)
+        repo.remotes["origin"].pull()
```

Comment:
If you're re-creating the tempdir when the script is run (if I'm reading https://github.com/mongodb/selected-tests/pull/9/files#diff-d17408d77ff47a5ff3f02d0ff1e7de90R65 correctly), you will always be cloning the repo, not pulling it, correct?

Reply:
Yup. I'll probably take out the if statement and just always pull explicitly

Reply:
Can you remove whichever branch of this if statement is not being used?